configure.ac

```diff
@@ -1300,10 +1300,25 @@ if test "$enable_fuzz" = "yes"; then
   enable_fuzz_binary=yes
 
   AX_CHECK_PREPROC_FLAG([-DABORT_ON_FAILED_ASSUME], [DEBUG_CPPFLAGS="$DEBUG_CPPFLAGS -DABORT_ON_FAILED_ASSUME"], [], [$CXXFLAG_WERROR])
+else
+  BITCOIN_QT_INIT
+
+  dnl sets $bitcoin_enable_qt, $bitcoin_enable_qt_test, $bitcoin_enable_qt_dbus
+  BITCOIN_QT_CONFIGURE([5.11.3])
 
+  dnl Keep a copy of the original $QT_INCLUDES and use it when invoking qt's moc
+  QT_INCLUDES_UNSUPPRESSED=$QT_INCLUDES
+  if test "$suppress_external_warnings" != "no" ; then
+    QT_INCLUDES=SUPPRESS_WARNINGS($QT_INCLUDES)
+    QT_DBUS_INCLUDES=SUPPRESS_WARNINGS($QT_DBUS_INCLUDES)
+    QT_TEST_INCLUDES=SUPPRESS_WARNINGS($QT_TEST_INCLUDES)
+  fi
+fi
+
+if test "$enable_fuzz_binary" = "yes"; then
   AC_MSG_CHECKING([whether main function is needed for fuzz binary])
   AX_CHECK_LINK_FLAG(
-    [-fsanitize=$use_sanitizers],
+    [-fsanitize= -fsanitize=$use_sanitizers], dnl Avoid re-use of an unrelated cached result.
```

Comment:
This looks like a hack, and the comment doesn't really explain why it's needed. If I understand what it's meant to be doing, couldn't you check that the `AC_LANG_PROGRAM` compiles while passing `SANITIZER_LDFLAGS` as extra flags?

Reply:
> This looks like a hack

It is a [hack](https://github.com/bitcoin/bitcoin/pull/24337#issuecomment-1046271814)...

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/24337#issuecomment-1089884069).